daemon/main.rs

```diff
@@ -36,6 +38,9 @@ async fn run(config: CliArgs) -> Result<()> {
         ));
     }
 
+    let state_file = PathBuf::from(config.state_root).join("state.json");
+    log::debug!("Using state file: {}", state_file.display());
```

Comment:
Discussion points:

- Are we fine with using JSON as the serialisation format? A binary format would be more performant, especially considering that we are (re)writing the contents of this file on every job completed.
- The state file will be stored in `STATE_ROOT/state.json`. My intention is to store module-specific states in subdirectories called `STATE_ROOT/{module_name}`, once we get there. I can imagine using `STATE_ROOT/runtime/state.json` for Zinnia state too.

Reply:
> * Are we fine with using JSON as the serialisation format? A binary format would be more performant, especially considering that we are (re)writing the contents of this file on every job completed.

I assume serialization will be <100ns, which would scale to >10k jobs per second, at which point disk performance has long become the bottleneck already. Therefore I think this is fine.

> * The state file will be stored in `STATE_ROOT/state.json`. My intention is to store module-specific states in subdirectories called `STATE_ROOT/{module_name}`, once we get there. I can imagine using `STATE_ROOT/runtime/state.json` for Zinnia state too.

👍